aiohttp/payload.py

```diff
@@ -827,10 +966,27 @@ async def write_with_length(
         except StopAsyncIteration:
             # Iterator is exhausted
             self._iter = None
+            self._consumed = True  # Iterator has been consumed
 
     def decode(self, encoding: str = "utf-8", errors: str = "strict") -> str:
         raise TypeError("Unable to decode.")
 
+    async def as_bytes(self, encoding: str = "utf-8", errors: str = "strict") -> bytes:
+        """Return bytes representation of the value.
+
+        This method reads the entire async iterable content and returns it as bytes.
+        It is equivalent to iterating through the async iterable and concatenating the results.
+        """
+        if self._iter is None:
+            return b""
+        chunks: List[bytes] = []
+        async for chunk in self._iter:
+            chunks.append(chunk)
+        # Iterator is exhausted after consuming
+        self._iter = None
+        self._consumed = True  # Iterator has been consumed
+        return b"".join(chunks)
```

Comment:
If we're going to load the entire thing into memory anyway, it could be used to make it re-consumable. Otherwise, there's probably no point to implementing this method.